src/library/basetracktablemodel.cpp

```diff
@@ -506,7 +506,11 @@ QVariant BaseTrackTableModel::roleValue(
             bool ok;
             const auto bpmValue = rawValue.toDouble(&ok);
             if (ok && bpmValue > 0.0) {
-                return QString("%1").arg(bpmValue, 0, 'f', 1);
+                if (fabs(round(bpmValue * 10) / 10 - bpmValue) < 0.001) {
```

Comment:
Extract this condition into a utility function with a descriptive name. Instead of returning a boolean it could return the number of decimal digits to simplify the code even more.

Reply:
I moved the rounding code into a static Bpm function